### PR TITLE
Changes since 0.5.3

### PR DESCRIPTION
**Pull request by xcompass**
This PR is includes the changes between 0.5.3 and 0.6 and used for Edx review. Here is the summary of the changes:

1. Allowed setup of a question with no correct answer.
2. Made the instructor-entered rationale optional for case where there is no correct answer.
3. Changed text in various locations to reflect option of scenario where there is no correct answer.
4. Inserted red asterisks in Studio view as appropriate to further indicate fields as required.
5. Fixed display of graphs on mobile.
6. Introduced a progress bar, with anchoring as appropriate, to better orient students during each stage of a question.
7. Changed colouring of both initial and final answers at last stage to correspond to correct answer, if specified, as appropriate. Initial and final answers are left black for the scenario where the instructor did not specify a correct answer.
8. Changed text labels to emphasize answer chosen by student, as appropriate.
9. Moved graphs headings above graphs for further emphasis.
10. Styled (italicized, emboldened) various labels for further emphasis.
11. Changed name of tool from Peer Instruction Tool to Peer Rationale Reflection Tool, and modified text in various locations as appropriate.

### CODE REVIEW

**Comment by xcompass** [comment]
@jleong-openedx please leave this PR open. This is just for edx review. We don't merge this as all the commits are already in 0.6

File: ubcpi/ubcpi.py [hunk 1]
```diff
@@ -117,6 +117,25 @@ def get_student_item_dict(self, anonymous_user_id=None):
...
+
+        Args:
+            anonymous_user_id(str): A unique anonymous_user_id for (user, course) pair.
+        Returns:
+            (dict): The student item associated with this XBlock instance. This
+                includes the student id, item id, and course id.
+        """
+
+        # This is not the real way course_ids should work, but this is a
+        # temporary expediency for LMS integration
```
Comment (jcdyer) [inline ubcpi/ubcpi.py]: This comment looks inaccurate.  course_ids aren't involved here.

File: ubcpi/static/js/src/d3-pibar.js [hunk 2]
```diff
@@ -10,12 +10,14 @@ d3.custom = (d3.custom || {});
...
  * ]
  */
-d3.custom.barChart = function() {
+d3.custom.barChart = function(scope) {
     // Private Variables
     var chartWidth  = 750;
     var chartHeight = 250;
     var minTotalFrequency = 10;
 
+    if(scope.role == 'instructor' || scope.role == 'staff'){ minTotalFrequency = 1};
```
Comment (jcdyer) [inline ubcpi/static/js/src/d3-pibar.js]: Where does this `.role` attribute get defined.  I see some code below, where there's a `$scope` getting `.user_role` assigned to it, but not `.role`.

File: ubcpi/static/js/src/ubcpi.js [hunk 3]
```diff
@@ -108,6 +108,9 @@ angular.module('UBCPI', ['ngSanitize', 'ngCookies'])
             $scope.options = data.options;
             $scope.rationale_size = data.rationale_size;
             $scope.weight = data.weight;
+            $scope.display_name = data.display_name;
+            $scope.user_role = data.user_role;
```
Comment (jcdyer) [inline ubcpi/static/js/src/ubcpi.js]: Here's the `$scope.user_role` mentioned above.

File: ubcpi/static/js/src/d3-pibar.js (hunk 2 above)
Comment (jleong-openedx) [inline ubcpi/static/js/src/d3-pibar.js]: In ubcpi-barchart-directive.js, this

role: '='

assigns the "role" attribute in ubcpi.html to the "role" scope property.

You can see the "role" attribute at the snippet

role="user_role">

in [ubcpi.html](https://github.com/ubc/ubcpi/blob/1f45d28508c6019fffd9af3857876ce32f09d4b2/ubcpi/static/html/ubcpi.html).

Comment (jcdyer) [inline ubcpi/static/js/src/d3-pibar.js]: Got it.

**Pull request merged**